mirdata/datasets/tonality_classicaldb.py

```diff
@@ -141,19 +141,19 @@ def __init__(self, track_id, data_home):
         self.title = self.audio_path.replace(".wav", "").split("/")[-1]
 
     @core.cached_property
-    def key(self) -> str:
+    def key(self) -> Optional[str]:
```

Comment:
Hey Dave, thanks for this! I have a naive/dumb question: `no-strict` means that `mypy` doesn't check the use of `None`, or am I confused? If that's what it does, why to use `Optional` here?

Reply:
It doesnt check the use of None. Mypy does not run at runtime. It analyzes the types passed in the editor to ensure inputs/outputs match. In this case the Use of Optional says that `key` will either return `str` or `None` (`Optional` is really `Union[str, None]`)

[strict](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-strict) in this case would apply a bunch of checks which are hard to run until types are spread to the rest of the code base.

`load_key(self.key_path)` returns an `Optional[str]` (courtesy of the coerce_to_string_io decorator). The typing was previously correct so this addition fixes the typing. Thanks to mypy I found the problem in the first place.

Reply:
I see, thanks @drubinstein!